src/Toml/Parser.hs

```diff
@@ -160,15 +161,30 @@ tableNameP :: Parser Key
 tableNameP = lexeme $ between (char '[') (char ']') keyP
 
 -- Values
-
 decimalP :: Parser Integer
-decimalP = mkInteger <$> decimalStringP
+decimalP = matchInteger >>= validateInteger
   where
-    decimalStringP   = fst <$> match (some digitChar >> many _digitsP)
```

Comment:
I want to say that I really like the previous implementation with `match` and `try`. I think it was quite efficient. Your implementation uses list of `Char`, but `match` relies on the efficient implementation of the underlying stream and uses of offsets. So I would like to have the implementation that behaves like the old one. But in this case it's kinda tricky to test, that the number is not started with zero but also not zero itself. For example, something like the following wouldn't work (but very close to the solution):

```haskell
mkInteger <$> notFollowedBy (some $ char '0') decimalStringP
```